apps/prairielearn/src/migrations/20231031163354_client_fingerprints__create.sql

```diff
@@ -0,0 +1,30 @@
+CREATE TABLE
+  client_fingerprints (
```

Comment:
We'll want at least one index for this table. I'm thinking `(user_id, session_id)` might be sufficient?

Reply:
I'm thinking we should add a `created_at TIMESTAMPTZ DEFAULT  now()` column to `client_fingerprints` and then index on `(user_id, session_id, created_at)` so we can rapidly find the most recent fingerprint for a given user with their current session.

Reply:
Per our conversation yesterday, we are now indexing on `(user_id, user_session_id, ip_address)`.